### PR TITLE
chore: remove `react-dropdowns.legacy`

### PR DESCRIPTION
**Pull request by ze-flo**
## Description

Removes `react-dropdowns.legacy`. Migration for `react-dropdowns.next` to be done in separate PR.

## Checklist

- [ ] ~~:ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)~~
- [ ] ~~:black_nib: copy updates are approved (add the content strategist as a reviewer)~~
- [ ] ~~:link: considered opportunities for adding cross-reference URLs (grep for keywords)~~
- [ ] ~~:wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver~~
- [ ] ~~:memo: tested in Chrome, Firefox, Safari, Edge~~

### CODE REVIEW

**Comment by ze-flo** [comment]
Reposting this [comment](https://github.com/zendeskgarden/website/pull/610#issuecomment-2269876949):

> Should we update the [migration guide](https://github.com/zendeskgarden/react-components/blob/1c9eabbb12b41b420410c23f1b925b20202ba6d4/docs/migration.md?plain=1#L116) to let our users know that while we're still publishing react-dropdown.legacy for compatibility purposes, its documentation will only be available in the V8 version of the website?

**Comment by geotrev** [comment]
> Reposting this [comment](https://github.com/zendeskgarden/website/pull/610#issuecomment-2269876949):
> 
> > Should we update the [migration guide](https://github.com/zendeskgarden/react-components/blob/1c9eabbb12b41b420410c23f1b925b20202ba6d4/docs/migration.md?plain=1#L116) to let our users know that while we're still publishing react-dropdown.legacy for compatibility purposes, its documentation will only be available in the V8 version of the website?

I think that's a reasonable think to do.

File: src/examples/menu/Nested.tsx [hunk 1]
```diff
@@ -9,8 +9,7 @@ import React, { useCallback, useState } from 'react';
 import styled from 'styled-components';
 import { hideVisually } from 'polished';
 import { Col, Row } from '@zendeskgarden/react-grid';
-import { Menu, Item, IItemProps } from '@zendeskgarden/react-dropdowns.next';
-import { Separator } from '@zendeskgarden/react-dropdowns';
```
Comment (geotrev) [inline src/examples/menu/Nested.tsx:13]: This must have been a weird mistake in the v8 docs. good catch.

File: src/examples/tooltip/TooltipPlacement.tsx [hunk 2]
```diff
@@ -6,12 +6,11 @@
...
 
 import React from 'react';
-import { Tooltip } from '@zendeskgarden/react-tooltips';
-import { GARDEN_PLACEMENT } from '@zendeskgarden/react-dropdowns';
+import { Tooltip, ITooltipProps } from '@zendeskgarden/react-tooltips';
 import { Row, Col } from '@zendeskgarden/react-grid';
 import { Button } from '@zendeskgarden/react-buttons';
 
-const PLACEMENTS: Record<string, GARDEN_PLACEMENT> = {
+const PLACEMENTS: Record<string, ITooltipProps['placement']> = {
```
Comment (geotrev) [inline src/examples/tooltip/TooltipPlacement.tsx:13]: I have this covered in #612 as well, but if yours merges first then I'll just rebase and it'll resolve automatically. :)

**Review by geotrev — CHANGES_REQUESTED** [review]
I want to say `src/fragments` can go (along with the `gatsby-config.ts` entry for that directory). Just double checking you saw that? 🙂

**Review by geotrev — APPROVED** [review]

**Comment by jzempel** [comment]
> Reposting this [comment](https://github.com/zendeskgarden/website/pull/610#issuecomment-2269876949):
> 
> > Should we update the [migration guide](https://github.com/zendeskgarden/react-components/blob/1c9eabbb12b41b420410c23f1b925b20202ba6d4/docs/migration.md?plain=1#L116) to let our users know that while we're still publishing react-dropdown.legacy for compatibility purposes, its documentation will only be available in the V8 version of the website?

The migration guide already spells out the existence of `dropdowns.legacy` and we won't have a final version 8 link artifact to the website until after v9 ships. I think we're good for now. If this becomes a repeat concern in the future, let's think about how to address it then (i.e. updating the in-place migration guide, etc)

**Pull request merged**